multichain-aggregator/multichain-aggregator-logic/src/repository/addresses.rs

```diff
@@ -159,6 +159,122 @@ where
     Ok(addresses)
 }
 
+pub async fn get_batch_in_order<C>(
+    db: &C,
+    pks: Vec<(&AddressAlloy, ChainId)>,
+) -> Result<Vec<Option<Model>>, DbErr>
+where
```

Comment:
can we simplify this function by changing result type to `HashMap<((Address, ChainId), Model)>`.
using this approach you can remove complicated logic with ordering by `Position`. also you wont return a lot of `None` in Vec if a lot of addresses is missing in database